### PR TITLE
General performance improvements

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
General performance improvements

## Changes
Fix re-compiling patterns, bad appends, useless string concat. Also done some small cleanup.

## Related Issues
N/A

## Testability

- [X] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ScriptHolder.java [hunk 1]
```diff
@@ -71,7 +72,7 @@ protected void exoticFarm(BlockMenu menu, Block block) {
 
 	public void openScript(Player p, Block b, String script) {
 		ChestMenu menu = new ChestMenu(ChatColor.DARK_AQUA + SlimefunPlugin.getLocal().getMessage(p, "android.scripts.editor"));
-		String[] commands = script.split("-");
+		String[] commands = PatternUtils.SLASH_SEPARATOR.split(script);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ScriptHolder.java]: SLASH? Shouldn't this be DASH?

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ScriptHolder.java [hunk 2]
```diff
@@ -391,7 +392,7 @@ protected String getScriptRatingPercentage(Config script) {
 
 	protected void openScriptComponentEditor(Player p, Block b, String script, int index) {
 		ChestMenu menu = new ChestMenu(ChatColor.DARK_AQUA + SlimefunPlugin.getLocal().getMessage(p, "android.scripts.editor"));
-		String[] commands = script.split("-");
+		String[] commands = PatternUtils.SLASH_SEPARATOR.split(script);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ScriptHolder.java]: Again: SLASH? Shouldn't this be DASH?

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/PatternUtils.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * often this can be a massive performance loss. This class solves that, one compile but many uses!
+ */
+public final class PatternUtils {
+
+    private PatternUtils() {}
+
+    public static final Pattern SEMICOLON = Pattern.compile(";");
+    public static final Pattern HASH = Pattern.compile("#");
+    public static final Pattern COMMA = Pattern.compile(",");
+    public static final Pattern SLASH_SEPARATOR = Pattern.compile(" / ");
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/PatternUtils.java]: Is this Slash safe here or does it need to be escaped?

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GearListener.java [hunk 4]
```diff
@@ -1,68 +1,57 @@
...
-		plugin.getServer().getPluginManager().registerEvents(this, plugin);
-	}
+    public GearListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onToggleSneak(PlayerToggleSneakEvent e) {
+        if (!e.isSneaking()) return;
+		Player p = e.getPlayer();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GearListener.java]: Please fix the indentation here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GearListener.java [hunk 5]
```diff
@@ -1,68 +1,57 @@
...
+
+        if (p.getInventory().getChestplate() != null) {
+            SlimefunItem chestplate = SlimefunItem.getByItem(p.getInventory().getChestplate());
+            if (chestplate == null || !Slimefun.hasUnlocked(p, chestplate, true)) return;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GearListener.java]: So if the Player wears no chestplate you return.
The jetboots will never even get checked if you return here prematurely.

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/PatternUtils.java (hunk 3 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/utils/PatternUtils.java]: yes its safe

**Comment by TheBusyBiscuit** [comment]
This PR has conflicting files btw. So you will need to re-pull it first.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 6]
```diff
@@ -138,7 +139,8 @@ public void onEnchant(EnchantItemEvent e) {
 			}
 			
 			String enchant = enchantments.get(random.nextInt(enchantments.size()));
-			e.getEnchantsToAdd().put(Enchantment.getByKey(NamespacedKey.minecraft(enchant.split("-")[0])), Integer.parseInt(enchant.split("-")[1]));
+			String[] enchantSplit = PatternUtils.SLASH_SEPARATOR.split(enchant);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java]: This should also rather be DASH than SLASH right?

Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java]: fixed

**Comment by TheBusyBiscuit** [comment]
Ehhhm sorry about that, you'll need to re-pull and merge one more time XD

**Comment by J3fftw1** [comment]
i hope the merge worked

**Pull request merged**